lauth/app/repositories/grant_repo.rb

```diff
@@ -11,11 +11,16 @@ def find(id)
       def for_user_and_uri(username, uri)
         ds = grants
           .dataset
-          .join(collections.name.dataset, uniqueIdentifier: :coll)
-          .join(locations.name.dataset, coll: :uniqueIdentifier)
-          .left_join(users.name.dataset, userid: grants[:userid])
-          .left_join(institution_memberships.name.dataset, inst: grants[:inst])
-          .left_join(group_memberships.name.dataset, user_grp: grants[:user_grp])
+          .where(grants[:dlpsDeleted] => "f")
+          .join(collections.name.dataset, uniqueIdentifier: :coll, dlpsDeleted: "f")
+          .join(locations.name.dataset, coll: :uniqueIdentifier) # , dlpsDeleted: "f") TODO: fix this
```

Comment:
Oh yeah, I did not have a test to check locations because of the factory traits thingy so I just TODO it for now.

Reply:
Ahh, right... I think we can apply what we learned with the grants: you can use a trait and still supply an explicit object for the association. However, this won't be quite right:

```
location = Factory[:location, :restricted_by_username]
collection = Factory[:collection, :restricted_by_username, locations: [location]]
```

There is a subtle problem here... When we define an association in the factory, the tuple/record is created. We have two factories defined, and each refers to the other: the location gets a parent collection and the collection gets a list of locations. Traits are referred to by symbol, so they cannot take a block or additional args -- basically just named prototypes. Whichever one we create first will create the other, and we can't change after the fact (without updating the database directly, as far as I can tell) or assign to attributes that are not declared.

It _may_ be possible to supply a block/proc as the value for an attribute and have that evaluated later... but I think the more direct answer is to make "deleted" traits for collections and locations, and use both.

That is, the second parameter to `Factory::[]` is a _list_ of traits, not a single trait. I'm sure we can add a `deleted` trait to the location and have some trait on collection use it. The question will be whether making a collection with two traits will combine the traits of the location association as we'd like...

But then again, we probably _do not_ want to imply the deletion of the location either way based on the deletion flag of the collection. We probably want to create (and test) all three additional combinations beyond the defaults:

- with a deleted location
- deleted with a non-deleted location
- deleted with a deleted location

And so, I think we probably should (or must) do something like:
```ruby
Factory[:collection, :restricted_by_username, :deleted]
# or
Factory[:collection, :restricted_by_username, :with_deleted_location]
# or
Factory[:collection, :restricted_by_username, :deleted, :with_deleted_location]
```
Even then, we might run into the problem that `restricted_by_username` declares the location association and we want to declare more about it (that it should use an additional trait). If that is the case, then we should probably spell out all four combinations as traits to be used individually on the collection factory.

These challenges are examples of why I prefer the builder pattern to the factory pattern in these scenarios -- being able to chain calls to refine the plan (requested object graph) before invoking the build makes this stuff easy to use and extend. But... this handles the ROM::Relations well and has been pretty reliable so far!

Reply:
fwiw, there are no deleted aa_coll_obj tuples